tests/components/synology_dsm/test_init.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for the Synology DSM component."""
+from homeassistant.components.synology_dsm import _async_setup_services
+from homeassistant.components.synology_dsm.const import DOMAIN, SERVICES
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .consts import HOST, PASSWORD, PORT, USE_SSL, USERNAME
+
+from tests.common import MockConfigEntry
+
+
+async def test_services_registered(hass: HomeAssistantType):
+    """Test if all services are registered."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: HOST,
+            CONF_PORT: PORT,
+            CONF_SSL: USE_SSL,
+            CONF_USERNAME: USERNAME,
+            CONF_PASSWORD: PASSWORD,
+        },
+    ).add_to_hass(hass)
+    await _async_setup_services(hass)
```

Comment:
Here, instead set up the integration with `hass.config_entries.async_setup`. See previous comment.

Reply:
Hi @MartinHjelmare ,

I have now changed to use `hass.config_entries.async_setup` but now I do not get it work ... it seems, that setup is not really done, because services where never registered 😕 
In addition I have patched `homeassistant.components.synology_dsm.SynoApi.async_setup` which should be the only call which causes I/O during setup, but still service registration does not work 😕 
Because my experience in writing tests with all these mock and patch and so on is nearly zero, I do not get it work by my own.
@Quentame could you please support in writing these test?

Reply:
We need to disable the default fixture in the conftest.py package module that is active for all tests in the synology_dsm test package.

https://github.com/home-assistant/core/blob/9a2142180780ba0c0d9da1857703f8ecae1b163f/tests/components/synology_dsm/conftest.py#L7-L13

Here's how to do that:
https://stackoverflow.com/a/38763328

Reply:
Hi @MartinHjelmare ,

thanks for your patients with me and the explanations 😃 
I figured out, that I have first to define a custom marker, before I can use the provided solution from https://stackoverflow.com/a/38763328

Reply:
Sorry @mib1185 but I have no idea on how to test this.
(I'm bad at mocking)

Reply:
No problem @Quentame, meanwhile it is solved 😃 
and yes, for me it is he same with writing tests and the logic in behind 🙈